lab_2_retrieval_w_bm25/main.py

```diff
@@ -18,6 +20,9 @@ def tokenize(text: str) -> list[str] | None:
 
     In case of corrupt input arguments, None is returned.
     """
+    if not isinstance(text, str):
+        return None
+    return re.sub(r'[^\s\w]+|\d+', r' ', text.lower()).split()
```

Comment:
is it working? can you explain this regular expression

Reply:
yes, it is working... 
this regular expression filters out symbols that are not a whitespace character or a letter. then they are replaced with a whitespace for each symbol in lowered text
[^] - everything but the expression in the brackets
\s - only whitespace characters
\w - only letters, digits and _
+ - one or more repetitions
| - or 
\d - only digits